modules/database/sqlite/load.php

```diff
@@ -0,0 +1,165 @@
+<?php
+/**
+ * Module Name: SQLite Integration
+ * Description: Use an SQLite database instead of MySQL. CAUTION: Enabling this module will bring up the WordPress installation screen. You will need to reconfigure your site, and you will lose all your data. If you then disable the module, you will get back to your previous MySQL database, with all your previous data intact.
+ * Experimental: Yes
+ *
+ * @package performance-lab
+ * @since n.e.x.t
+ */
+
+// Temporary - This will be in wp-config.php once SQLite is merged in Core.
+if ( ! defined( 'DATABASE_TYPE' ) ) {
+	define( 'DATABASE_TYPE', 'sqlite' );
+}
+
+/**
+ * Adds the db.php file in wp-content.
+ *
+ * When the plugin gets merged in wp-core, this is not to be ported.
+ *
+ * @since n.e.x.t
+ */
+function perflab_sqlite_plugin_copy_db_file() {
+	// Bail early if the SQLite3 class does not exist.
+	if ( ! class_exists( 'SQLite3' ) ) {
+		return;
+	}
+
+	$destination = WP_CONTENT_DIR . '/db.php';
+
+	// Bail early if the file already exists.
+	if ( file_exists( $destination ) ) {
+		return;
+	}
+
+	// Init the filesystem to allow copying the file.
+	global $wp_filesystem;
+	if ( ! $wp_filesystem ) {
+		require_once ABSPATH . '/wp-admin/includes/file.php';
+		WP_Filesystem();
+	}
+
+	$file_copied_successfully = false;
+	if ( $wp_filesystem->touch( $destination ) ) {
+
+		// Get the db.copy file contents, replace placeholders and write it to the destination.
+		$file_contents = str_replace(
+			array(
+				'{SQLITE_IMPLEMENTATION_FOLDER_PATH}',
+				'{PERFLAB_PLUGIN}',
+				'{SQLITE_MODULE}',
+				'{PERFLAB_MODULES_SETTING}',
+			),
+			array(
+				__DIR__,
+				str_replace( WP_PLUGIN_DIR . '/', '', PERFLAB_MAIN_FILE ),
+				'database/sqlite',
+				PERFLAB_MODULES_SETTING,
+			),
+			file_get_contents( __DIR__ . '/db.copy' )
+		);
+
+		$file_copied_successfully = $wp_filesystem->put_contents( $destination, $file_contents );
+	}
+
+	if ( $file_copied_successfully ) {
+		/**
+		 * This hack is here because it's impossible to redirect properly in PHP,
+		 * due to hooks race conditions and the fact that the redirect is done before the output is sent.
+		 *
+		 * When SQLite gets merged in Core this will be a non-issue.
+		 */
+		add_action(
+			'admin_head',
+			function() {
+				echo '<script>window.location.reload(true);</script>';
+			}
+		);
+	}
+}
+add_action( 'plugins_loaded', 'perflab_sqlite_plugin_copy_db_file' );
```

Comment:
Following up on https://github.com/WordPress/performance/pull/547#discussion_r1003797150, makes sense that this wouldn't get executed here on module activation.

With that said, this shows that we should have a proper solution for this. I would suggest the following:
* Add a filter callback for `'pre_update_option_' . PERFLAB_MODULES_SETTING` to the main plugin `load.php` file.
* In that callback, do two things:
    * Iterate through all module keys which are now active but before were inactive and potentially run module activation routine:
        * Check if the module directory has an `activation.php` file.
        * If so, it should be expected to return a single closure for the activation logic (e.g. similar to the `can-load.php` file returns a closure).
        * Then, call the function.
    * Iterate through all module keys which are now inactive but before were active and potentially run module deactivation routine:
        * Check if the module directory has an `deactivation.php` file.
        * If so, it should be expected to return a single closure for the deactivation logic (e.g. similar to the `can-load.php` file returns a closure).
        * Then, call the function.

This way we would have a clean solution in which a module could provide its activation and deactivation logic, and then you could move this function here, and the one below, to these new `activation.php` and `deactivation.php` files for the new SQLite module.

Reply:
Great idea! Done 👍